frontend/lib/solidus_frontend.rb

```diff
@@ -1,3 +1,12 @@
 # frozen_string_literal: true
 
+require 'spree/deprecation'
+
+deprecation_message = <<~MSG
+  NOTE: SolidusFrontend is deprecated; use SolidusStarterFrontend instead.
```

Comment:
I think this message is not clear. We don't want people currently using `solidus_frontend` to start using the new frontend because they would have to rebuild their frontend entirely. I think what we want to say to people is that, this gem will be removed from the meta-package `solidus` and if they still want to use it, they need to add it manually to the Gemfile, isn't it?

Reply:
@kennyadsl How about this:

> NOTE: SolidusFrontend is deprecated. It will be moved out of the `solidus` meta-package gem in a future version of Solidus. Once moved out, you'll need to manually add `solidus_frontend` to your Gemfile in order to continue using it.

> For fresh Solidus applications, we recommend you use SolidusStarterFrontend instead.

Reply:
I think the text of the message may work. I'm only a bit concerned that, with this approach, there's no way to remove the deprecation message until we actually remove `solidus_frontend` from `solidus`. Is there a way to check that solidus_frontend is present in the Gemfile and print the message only otherwise? Or do you have any other idea about how to fix this?

Reply:
> Is there a way to check that solidus_frontend is present in the Gemfile and print the message only otherwise?

That question intrigued me :slightly_smiling_face: It seems we can get the information through:

```ruby
Bundler.locked_gems.dependencies
```

However, it seems to be very slow. I don't know if there's a way to cache it thanks to the `bundler/setup` Rails apps do in `config/boot.rb`

Reply:
and probably move at the boot phase so that, even if it's slow it only happens once when the application boots (but maybe it's the same in this file?).

Reply:
> I'm only a bit concerned that, with this approach, there's no way to remove the deprecation message until we actually remove solidus_frontend from solidus. 

I'm currently extracting `solidus_frontend` to an external gem. By pointing `solidus_frontend` in the Gemfile to that external gem, we'd be able to remove the deprecation message.

To demonstrate, please see the Solidus app in https://github.com/gsmendoza/external_solidus_frontend_store. In that app's [Gemfile](https://github.com/gsmendoza/external_solidus_frontend_store/blob/main/Gemfile#L62), all the Solidus gems except for `solidus_frontend` point to this branch. In contrast, `solidus_frontend` points to the `solidus_frontend` repo I'm working on: https://github.com/gsmendoza/solidus_frontend.

If we run that app's server, then it wouldn't load the `solidus_frontend` in this Solidus meta package, and as such, wouldn't display the deprecation message.

Does this resolve the concern you have @kennyadsl?

Reply:
@gsmendoza, I'm not sure I follow. `solidus_frontend` is already a [standalone gem](https://rubygems.org/gems/solidus_frontend). Then, it's a dependency on [solidus meta-gem](https://rubygems.org/gems/solidus).

Reply:
Yes, this does solve! 🙂 

You are right, if people switch to the gem in the Gemfile they will be able to get the version without the deprecation notice. There's just one thing to address: we need to understand how the versioning of those two gems will work, because both need to be on RubyGems, and they can't have the same version.

Reply:
@waiting-for-dev we can move `solidus_frontend` to another GitHub repository, remove it as a dependency of `solidus`, and start publishing new versions on RubyGems of it from its own new repository only. Do you see any issue with this approach?

Reply:
Oh, I get it. Yeah, makes sense :slightly_smiling_face:

Reply:
> we need to understand how the versioning of those two gems will work, because both need to be on RubyGems, and they can't have the same version.

@kennyadsl I remember your idea before is to give the external gem a different name e.g. `solidus_deprecated_frontend`. If we do that, we can maintain both `solidus_frontend` and `solidus_deprecated_frontend` in RubyGems :)